packages/ra-core/src/controller/useList.ts

```diff
@@ -241,7 +241,7 @@ export const useList = (props: UseListOptions): UseListValue => {
         setPerPage,
         setSort,
         showFilter,
-        total: finalItems.ids.length,
+        total: ids.length,
```

Comment:
no, the total should take into account the filtering, so you need to introduce a new variable to get the length after filterring but before pagination